### ISSUE TITLE
Clarification about espeak.js async callbacks usage

### ISSUE THREAD

**Issue by animaone** [espeak-ng/espeak-ng#840]
Hello. I have compiled espeak.js and I am trying to modify the demo so I can get an IPA text from a console function, but the result.ipa variable from demo.js does not become available. there is something special in document.getElementById('ipaarea').value = result.ipa; that forces the async callback to return a correct ipa text value instead of 'undefined'?
**related to file: emscripten/js/demo.js**

**Comment by animaone** [comment]
Ok... the problem is not in espeak.
It is about context access, since Webworkers can't access the context of the main thread, I have managed to serialize the object I want access to into an Array, and than passed this object's index to the context of tts.synthesize_ipa.

I also modified the function synthesize_ipa a bit, so that it can work in this context. The function can receive more than one argument, and the argument format is: **tts.synthesize_ipa({text: text, index: i}, ipaCallback);**

I also have edited  synthesize_ipa, because I don't know how to implement this in the emscripten glue. It would be nice if someone could do that.

function synthesize_ipa_override(callerArguments, aCallback) {
	aText = callerArguments.text;
    var ipaVirtualFileName = "espeak-ng-ipa-tmp-" + Math.random().toString().substring(2);
    var res = "";
    var code = this.synth_ipa_(aText, ipaVirtualFileName);
    if (code == 0) res = FS.readFile(ipaVirtualFileName, {encoding: "utf8"});
    FS.unlink(ipaVirtualFileName);
    var ret = {code: code, ipa: res, callerArguments: callerArguments, mytest:"this is an override"};
    return ret
}
